### PR TITLE
Add a vehicle name tag

### LINKED ISSUE #2182: Feature suggestion: add vehicle name parameter
I would like to propose to add a vehicle name parameter, that makes it easier to remember the vehicle that a certain log belongs to. 

It would also allow to auto-generate flight logbooks (as required in some legislations for UAVs) from dataflash logs via simple scripting.

### PR DESCRIPTION
**Pull request by hsteinhaus**
I consider this PR more or less as RFC, as I am not sure if I have already found the best possible solution for storing fixed-length string params. Not tested yet (not too easy to do without modifiying at lesat one GCS first).

Fixes #2182

### CODE REVIEW

**Comment by magicrub** [comment]
Wow, that is a much more elegant solution than what I was thinking of. Nice one!

**Comment by hsteinhaus** [comment]
Unfortunately I can not reproduce the error/warning that is Travis complaining about locally. Any ideas where the problem is?

File: libraries/AP_Param/AP_Param.h [hunk 1]
```diff
@@ -62,7 +65,8 @@ enum ap_var_type {
     AP_PARAM_VECTOR3F,
     AP_PARAM_VECTOR6F,
     AP_PARAM_MATRIX3F,
-    AP_PARAM_GROUP
+    AP_PARAM_GROUP,
+    AP_PARAM_STR8,
```
Comment (magicrub) [inline libraries/AP_Param/AP_Param.h]: does this extra comma mess things up?

**Comment by tridge** [comment]
unfortunately the current MAVLink protocol can only transport float values, so this won't be possible without some major rework of our MAVLink code

**Comment by magicrub** [comment]
So @tridge, is the MAVLink limitation a killer for this PR? Shall we close?

**Comment by hsteinhaus** [comment]
I just noticed that the Tower/Droneshare combo has the same problems of assigning logs to vehicles.

While DroneShare supports more than one vehicle, Tower has no choice but to create a dummy vehicle called "quadcopter". It then assigns all recorded logs to this vehicle. Not really a nice solution IMHO.

**Comment by tridge** [comment]
as all parameter updates go via 4 byte floats this PR really won't work.
We can try something different if/when we move to a new parameter mavlink protocol that can support strings

**Comment by magicrub** [comment]
Note, there is at least a BRD_SERIAL_NUM param in matter now. Baby step.

**Pull request closed without merging**